lcls_naming_tool/lcls_naming_tool.py

```diff
@@ -0,0 +1,244 @@
+    
+    3. To check if a list of PV names is valid pipe in the file name like so 
+            $ cat pvlist.txt | ./lcls_naming_tool.py
+'''
+
+
+import sys
+import json
+import re
+
+
+fc_valid = True
+fg_valid = True
+ccc_valid = True
+nn_valid = True
+cspv_valid = True
+fc_dict = {}
+fg_dict = {}
+ccc_dict = {}
+beam_sources = {'K': '', 'L': ''}
+beam_numbers = {'0': '', '1': '', '2': '', '3': '', '4': '', '5': ''}
+
+
+def load_taxons():
+    global fc_dict
+    global fg_dict
+    global ccc_dict
+
+    # Read the json files containing all the taxons
+    with open('functional_component_taxon.json') as fc_file:
+        fc_dict = json.load(fc_file)
+
+    with open('fungible_element_taxon.json') as fg_file:
+        fg_dict = json.load(fg_file)
+
+    with open('ccc_taxon.json') as ccc_file:
+        ccc_dict = json.load(ccc_file)
+
+
+def functional_component_is_valid(fc_taxon):
+    
+    # check length of the functional component
+    if (len(fc_taxon) == 5):
+        fc_prefix = str(fc_taxon[0] + fc_taxon[1])
+        fc_source_ltr = str(fc_taxon[3])
+        fc_beam_num = str(fc_taxon[4])
+    else:
+        return False
+
+    # validate FFFFF
+    try:
+        fc_dict[fc_prefix]
+        beam_sources[fc_source_ltr]
+        beam_numbers[fc_beam_num]
+
+    except KeyError:
+        return False
+
+    else:
+        return True
+
+
+def constituent_component_is_valid(ccc_taxon):
+    try:
+        ccc_dict[ccc_taxon]
+    except:
+        return False
+    else:
+        return True
+
+
+def fungible_is_valid(fg_taxon):
+    try:
+        fg_components = fg_taxon.replace(
+            '_', ' ').replace(':', ' ').split()
+
+        for name in fg_components:
+            fg_dict[name]
+
+    except KeyError:
+        return False
+
+    else:
+        return True
+
+
+def increment_is_valid(nn_taxon):
+    # check the increment is an int and has at least 2 digits
+    try:
+        assert int(nn_taxon)
+        assert len(nn_taxon) >= 2
+
+    except ValueError:
+        return False
+
+    else:
+        return True
+    
+
+def process_variable_is_valid(cspv_taxon):
+    # check control system process variable starts with a letter
+    match = re.search('^[a-zA-z]', cspv_taxon)
+    if match:
+        return True
+    else:
+        return False
+
+
+def validate(user_input):
+    try:
+        assert '.' not in user_input
+        assert (len(user_input) <= 60)
+
+    except AssertionError:
+        print('Invalid')
+        return False
+
+    pv_name = [x.strip() for x in user_input.split(':')]
+
+    for element in pv_name:
+        if not re.match('^[a-zA-Z0-9]+', element):
+            print('Invalid')
+            return False
+
+    # Check the length of the PV name is valid
+    if (len(pv_name) < 3) or (len(pv_name) > 5):
+        print('Invalid')
+        return False
+
+    # parse the functional component (FFFFF)
+    fc_taxon = list(pv_name[0].strip())
```

Comment:
I think some of the other implementations get easier if this is a str instead of a list of str

Reply:
I agree, python strings already implement most of the list interface so you can use similar manipulation methods